src/main/java/org/opensearch/knn/index/query/filtered/KNNQueryVector.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.query.filtered;
+
+import org.apache.lucene.index.BinaryDocValues;
+import org.opensearch.knn.index.SpaceType;
+
+import java.io.IOException;
+
+/**
+ * Wrapper interface on knn query vector to hide its type(float, byte) and provide score for given doc value and space type
+ */
+public interface KNNQueryVector {
```

Comment:
I still think we should avoid this interface atleast from the scope of this PR and think through this is more details when we have a other types of vectors with Native engines. Because that time, we will sweep through all the classes of K-NN and will ensure that they are able to support both floats and non float vectors.

Reply:
ack